Firmware/ARYTHMATIK_Euclid/ARYTHMATIK_Euclid.ino

```diff
@@ -398,133 +600,642 @@ void loop() {
       }
     }
   }
+  
+  if (gate_timer + 10 <= millis()) {  //off all gate , gate time is 10msec
+
+    FastGPIO::Pin<5>::setOutput(0);
+    FastGPIO::Pin<6>::setOutput(0);
+    FastGPIO::Pin<7>::setOutput(0);
+    FastGPIO::Pin<8>::setOutput(0);
+    FastGPIO::Pin<9>::setOutput(0);
+    FastGPIO::Pin<10>::setOutput(0);
+  }
+  if (gate_timer + 30 <= millis()) {  //off all gate , gate time is 10msec, reduced from 100 ms to 30 ms
+    FastGPIO::Pin<4>::setOutput(0);
+    FastGPIO::Pin<14>::setOutput(0);
+    FastGPIO::Pin<15>::setOutput(0);
+    FastGPIO::Pin<16>::setOutput(0);
+    FastGPIO::Pin<17>::setOutput(0);
+    FastGPIO::Pin<0>::setOutput(0);
+    FastGPIO::Pin<1>::setOutput(0);
+  }
+
+  if (disp_refresh == 1) {
+    OLED_display();  //refresh display
+    disp_refresh = 0;
+  }
+}
+
+
+
+void factoryReset() {
+  initializeDefaultRhythms();  // Re-initialize EEPROM with default rhythms
+  // Indicate on display or via LED that factory reset is complete
+}
 
-  if  (gate_timer + 10 <= millis()) { //off all gate , gate time is 10msec
 
-    digitalWrite(5, LOW);
-    digitalWrite(6, LOW);
-    digitalWrite(7, LOW);
-    digitalWrite(8, LOW);
-    digitalWrite(9, LOW);
-    digitalWrite(10, LOW);
+void saveToEEPROM(int slot) {
+  int baseAddress = EEPROM_START_ADDRESS + (slot * CONFIG_SIZE);
+
+  for (int ch = 0; ch < 6; ++ch) {
+    EEPROM.put(baseAddress + ch * sizeof(byte), memorySlots[slot].hits[ch]);        // hits
```

Comment:
Why not store the whole struct? Then on load pick the request channel from the full loaded value.
```c
EEPROM.put(0, memorySlots);
```

Reply:
done.